### PR TITLE
Add initial doc templates for theming and a script for applying them.

### PR DESCRIPTION
**Pull request by jverkoey**
The added script allows a component to apply the template in an automated fashion.

```
Usage: ./scripts/apply_template <ComponentName> <path_to_.template_file>

Copies the template into the component directory and replaces all of the symbols defined
in the component's .vars file.

.vars files should be placed in the root of the component's directory.
Example: components/ActivityIndicator/.vars

Each line of the .vars file should be of the form symbol=replacement.
Example:
component=ActivityIndicator
component_name=Activity Indicator

This script will replace all instances of <#symbol#> in the .template with the given
replacement.

Example usage: ./scripts/apply_template ActivityIndicator scripts/templates/component/docs/ColorTheming.md.template
```

### CODE REVIEW

File: docs/templates/component/docs/ColorTheming.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+``` bash
+pod 'MaterialComponents/<#component#>+Extensions/ColorThemer'
+```
+
+You can then import the theming APIs:
+
+<!--<div class="material-code-render" markdown="1">-->
+#### Swift
+``` swift
+import MaterialComponents.Material<#component#>_ColorThemer
```
Comment (yarneo) [inline docs/templates/component/docs/ColorTheming.md]: I feel like this red is very intrusive, could we have it the same color scheme as the other wildcards?

File: docs/templates/component/docs/ColorTheming.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+```
+<!--</div>-->
+
+<#themer_api#> allows you to theme a <#component_name#> with your app's color
+scheme.
+
+<!--<div class="material-code-render" markdown="1">-->
+#### Swift
+``` swift
+let colorScheme: MDCSemanticColorScheme()
```
Comment (yarneo) [inline docs/templates/component/docs/ColorTheming.md]: `let colorScheme = MDCSemanticColorScheme()`

File: docs/templates/component/docs/TypographyTheming.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+```
+<!--</div>-->
+
+<#themer_api#> allows you to theme a <#component_name#> with your app's typography
+scheme.
+
+<!--<div class="material-code-render" markdown="1">-->
+#### Swift
+``` swift
+let typographyScheme: MDCSemanticTypographyScheme()
```
Comment (yarneo) [inline docs/templates/component/docs/TypographyTheming.md]: `:` -> `=`

**Review by yarneo — DISMISSED** [review]

File: docs/templates/component/docs/ColorTheming.md (hunk 1 above)
Comment (jverkoey) [reply docs/templates/component/docs/ColorTheming.md]: To be honest I'm not sure why it's like this.

File: docs/templates/component/docs/ColorTheming.md (hunk 2 above)
Comment (jverkoey) [reply docs/templates/component/docs/ColorTheming.md]: Done.

File: docs/templates/component/docs/TypographyTheming.md (hunk 3 above)
Comment (jverkoey) [reply docs/templates/component/docs/TypographyTheming.md]: Done.

**Comment by jverkoey** [comment]
Example of the template applied to ActivityIndicator:

screen shot 2018-04-27 at 10 36 03 am

**Review by mohammadcazig — APPROVED** [review]
LGTM

**Pull request merged**